source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts

```diff
@@ -0,0 +1,208 @@
+import { KinesisFirehoseToS3 } from '@aws-solutions-constructs/aws-kinesisfirehose-s3';
+import { LambdaToKinesisFirehose } from '../lib';
+import * as defaults from '@aws-solutions-constructs/core';
+
+test('Test that properties are set correctly', () => {
+  // Stack
+  const stack = new cdk.Stack();
+  const destination = GetTestDestination(stack, 'test-destination');
+
+  const construct = new LambdaToKinesisFirehose(stack, 'test-target', {
+    existingKinesisFirehose: destination.kinesisFirehose,
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: "index.handler",
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    }
+  });
+
+  expect(construct.kinesisFirehose).toBeDefined();
+  expect(construct.lambdaFunction).toBeDefined();
+  expect(construct.vpc).not.toBeDefined();
+});
+
+test('Test existing function is utilzed correctly', () => {
+  // Stack
+  const stack = new cdk.Stack();
+  const destination = GetTestDestination(stack, 'test-destination');
+  const testName = 'test-name';
+  const existingFunction = new lambda.Function(stack, 'test-function', {
+    runtime: lambda.Runtime.NODEJS_18_X,
+    handler: "index.handler",
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    functionName: testName
+  });
+
+  new LambdaToKinesisFirehose(stack, 'test-target', {
+    existingKinesisFirehose: destination.kinesisFirehose,
+    existingLambdaObj: existingFunction
+  });
+
+  expect(stack).toHaveResource('AWS::Lambda::Function', {
+    FunctionName: testName
+  });
+
+  // This is 2 because there's a lambda function in the custom resource to
+  // delete all the objects when cleaning up the s3 bucket in kinesisfirehose-s3
+  expect(stack).toCountResources('AWS::Lambda::Function', 2);
+});
+
+test('Test that lambda function props are incorporated', () => {
+  // Stack
+  const stack = new cdk.Stack();
+  const destination = GetTestDestination(stack, 'test-destination');
+  const testName = 'test-name';
+
+  new LambdaToKinesisFirehose(stack, 'test-target', {
+    existingKinesisFirehose: destination.kinesisFirehose,
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: "index.handler",
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+      functionName: testName
+    }
+  });
+
+  expect(stack).toHaveResource('AWS::Lambda::Function', {
+    FunctionName: testName
+  });
+});
+
+test('Test that new VPC is created', () => {
+  // Stack
+  const stack = new cdk.Stack();
+  const cidrRange = '172.0.0.0/16';
+
+  const destination = GetTestDestination(stack, 'test-destination');
+
+  const construct = new LambdaToKinesisFirehose(stack, 'test-target', {
+    existingKinesisFirehose: destination.kinesisFirehose,
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: "index.handler",
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    },
+    deployVpc: true,
```

Comment:
What happens if deployVpc is false or undefined, and the user still includes `vpcProps`? Do we expect the construct to still use them, or do we throw an error?